backend/src/schema/resolvers/users.js

```diff
@@ -11,5 +11,27 @@ export default {
       params = await fileUpload(params, { file: 'avatarUpload', url: 'avatar' })
       return neo4jgraphql(object, params, context, resolveInfo, false)
     },
+    DeleteUser: async (object, params, context, resolveInfo) => {
+      const { resource } = params
+      const session = context.driver.session()
+
+      if (resource && resource.length) {
+        await Promise.all(
+          resource.map(async node => {
+            await session.run(
+              `
+            MATCH (resource:${node})<-[:WROTE]-(author:User {id: $userId})
+            DETACH DELETE resource
+            RETURN author`,
+              {
+                userId: context.user.id,
+              },
+            )
+          }),
+        )
+        session.close()
```

Comment:
Most importantly is that if you delete the nodes, let's say you delete a post, then you won't delete all the comments on that post, because they belong to a different user. So my suggestion is not to delete a post but delete **all attributes** of that post.

I would say it's good practice to have something like a tombstone to maintain the context and the associations with the data that is already there. This is even more important with a comment. If you have a conversation with deleted comments and you cannot see anymore where a comment has been deleted, the responses to that comment might look awkward and out of context.

Reply:
hey @roschaefer, I think I followed all of your advice, or at least tried to...
I agree that it would be good to have some "tombstone", if a user deletes their comment, either by deleting their account, with all its comments, or by deleting a single comment.
It feels to me that this functionality is a separate ticket though, don't you agree? 
The reason I feel like that is cause the simple action of deleting one comment doesn't support it, much less deleting all comments.

Reply:
Maybe when a user deletes a comment they have made, instead of deleting the node, we could delete all attributes, except the `id`, and change the content to `this comment has been deleted by the user`, plus add a `deletedOn` date?

Reply:
Yes, that is the implementation of the alpha! We currently save a boolean flag on nodes `deleted = true`. Setting a date would be better, but we have to change the implementation of all the counters etc. I would suggest to open another issue for refactoring the `deleted` and `disabled` flags from type Boolean to type DateTime.

Reply:
So, you suggesting that instead of deleting the comment, we should be just changing the attribute `delete = true`?

And for `posts`, you would expect that we not delete them completely because there are other user's comments on them? just delete the user relationship and it's content??

Reply:
Regarding the comments, yes. Regarding the posts: Up to you. You won't see deleted posts in your newsfeed. So basically removing the posts will certainly make the posts along with the comments unlocatable. There will be probably no difference from the end user perspective, however we administrators could make use of the data if the relationships are still there at least.

Reply:
taken care of in the latest commit